src/server/ocpp/services/OCPPService.ts

```diff
@@ -714,8 +744,18 @@ export default class OCPPService {
       await OCPPUtils.priceTransaction(headers.tenantID, transaction, consumption, TransactionAction.START);
       // Billing
       await OCPPUtils.billTransaction(headers.tenantID, transaction, TransactionAction.START);
-      // OCPI
-      await OCPPUtils.processOCPITransaction(headers.tenantID, transaction, chargingStation, TransactionAction.START);
+      // Roaming ?
+      if (transaction.user || !transaction.user.issuer) {
```

Comment:
That's a problem I had. Between the authorization and the transaction start the information about the authorization is lost.
The transaction object will be created in the handleStartTransaction method.

At the time when the authorization happens there is no transaction object to store the information about the roaming plattform or authorization method.

For OICP there come more problems with that information loss. I need information about the authorization method and session id that Hubject created after a successful authorization. At the moment I put the information in a RemoteAuthorization object that is stored in the charging station. That works but I would also prefer a cleaner solution.

Reply:
The authorisation information should be put at the transaction level like it's already the case for the tag authorisation scheme.  
Either as an object directly attached to the transaction or inside a dedicated collection for roaming authorisation.
Since you already have done the code to put it inside an object, you can just put it inside the oicp extension for transaction object.